workspaces/ui-v2/src/pages/changelogHistory/ChangelogHistory.tsx

```diff
@@ -1,23 +1,32 @@
-import React, { FC } from 'react';
-import { makeStyles } from '@material-ui/core/styles';
+import React, { FC, useState } from 'react';
+import { Button, makeStyles } from '@material-ui/core';
 import { Schedule as ScheduleIcon } from '@material-ui/icons';
-
-import { Button } from '@material-ui/core';
+import { useHistory } from 'react-router-dom';
 
 import {
   Page,
   useChangelogPages,
   useDocumentationPageLink,
 } from '<src>/components';
-import { useBatchCommits } from '<src>/hooks/useBatchCommits';
+import { BatchCommit, useBatchCommits } from '<src>/hooks/useBatchCommits';
 import { formatTimeAgo } from '<src>/utils';
 
+import { ConfirmResetModal } from './components';
+
+const SPECIFICATION_METADATA_COMMIT_MESSAGE =
```

Comment:
This needs to match up with the spectacle commit message (this is probably the "best" way to identify this commit). There's downsides (obviously) with using the commit message - we could also make the assumption the first commit is unrevertable; which for most cases first commit === specification metadata commit. 

The longer term thing here is the metadata probably should live outside of the temporal spec file

Reply:
Uhm, isn't this a reset _to_ the selected commit, making the selected commit the new "current" commit? That should make the initial commit a fine candidate as a reset target. Once it becomes the current, it can no longer be reset back any further.

Reply:
Fixed in https://github.com/opticdev/optic/pull/919/commits/448903fb3a776eaf66145ea9319d440b8c5d29aa: turns out you can pass a `component` prop to a `Button`, a feature that seems to have been added for [exactly this purpose](https://material-ui.com/components/buttons/#third-party-routing-library).